### PR TITLE
Pagination 

### PR DESCRIPTION
**Pull request by Fran-A-Dev**
## Tasks

- [ Yes] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.

## Description
Pagination Guide

### CODE REVIEW

File: internal/faustjs.org/docs/guides/pagination.mdx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+The Apollo client can implement [relay-style pagination](https://www.apollographql.com/docs/react/pagination/cursor-based/) with the relay spec using merge and read functions, which means all the details of `connections`, `edges` and `pageInfo` can be abstracted away, into a single, reusable helper function. WPGraphQL follows the relay spec as well.
+We are importing the `relayStylePagination` function from Apollo and then a `class component` is created which is the basic syntax of the Faust plugin.
+
+Next, we have an `apply` method with the `hooks` parameter which is an object that gives you a function called `addFilter`. The `addFilter` function allows us to modify the [Apollo Client’s `InMemoryCacheOptions`](https://www.apollographql.com/docs/react/caching/cache-configuration/).
+
+The `addFilter` hook is taken and calls the memory cache function options. The options are coming from the Apollo Client Cache configuration. These options allow for configuring the cache for  pagination. The faust namespace follows that.
+
+## Register the Plugin for Pagination
+
+Import and register your plugin for pagination in `faust.config.js`
```
Comment (blakewilson) [inline internal/faustjs.org/docs/guides/pagination.mdx:51]:
```suggestion
Import and register your plugin for pagination in `faust.config.js` by adding it to the `experimentalPlugins` array:
```

File: internal/faustjs.org/docs/guides/pagination.mdx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+This guide covers how to set up Pagination in your Faust.js site with Apollo and WPGraphQL.
+
+## Setup
+
+To get started, create a folder in the root of your Faust.js project called `plugins`.  In this `plugins` folder, create a file called `RelayStylePagination.js` and insert this content:
+
+```
+import { relayStylePagination } from "@apollo/client/utilities";
+
+export class RelayStylePagination {
```
Comment (blakewilson) [inline internal/faustjs.org/docs/guides/pagination.mdx:16]:
```suggestion
export class RelayStylePaginationPlugin {
```

Comment (blakewilson) [reply internal/faustjs.org/docs/guides/pagination.mdx:16]: Let's name it including "plugin" just to be more clear.

File: internal/faustjs.org/docs/guides/pagination.mdx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## Register the Plugin for Pagination
+
+Import and register your plugin for pagination in `faust.config.js`
+
+```
+import { setConfig } from "@faustwp/core";
+import templates from "./wp-templates";
+import possibleTypes from "./possibleTypes.json";
+import { RelayStylePagination } from "./plugins/RelayStylePagination";
```
Comment (blakewilson) [inline internal/faustjs.org/docs/guides/pagination.mdx:57]:
```suggestion
import { RelayStylePaginationPlugin } from "./plugins/RelayStylePagination";
```

File: internal/faustjs.org/docs/guides/pagination.mdx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+Next, we have an `apply` method with the `hooks` parameter which is an object that gives you a function called `addFilter`. The `addFilter` function allows us to modify the [Apollo Client’s `InMemoryCacheOptions`](https://www.apollographql.com/docs/react/caching/cache-configuration/).
+
+The `addFilter` hook is taken and calls the memory cache function options. The options are coming from the Apollo Client Cache configuration. These options allow for configuring the cache for  pagination. The faust namespace follows that.
+
+## Register the Plugin for Pagination
+
+Import and register your plugin for pagination in `faust.config.js`
+
+```
```
Comment (blakewilson) [inline internal/faustjs.org/docs/guides/pagination.mdx:53]:
```suggestion
```js
```

File: internal/faustjs.org/docs/guides/pagination.mdx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+export default setConfig({
+  templates,
+  experimentalPlugins: [new RelayStylePagination()],
+  possibleTypes,
+});
+```
+
+## Example WPGraphQL Pagination Query
+
+```
```
Comment (blakewilson) [inline internal/faustjs.org/docs/guides/pagination.mdx:71]:
```suggestion
```graphql
```

File: internal/faustjs.org/docs/guides/pagination.mdx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+          slug
+        }
+      }
+    }
+  }
+`;
+```
+This is an example of a query with the fields for pagination being `hasNextPage` and `endCursor`.
+
+
```
Comment (blakewilson) [inline internal/faustjs.org/docs/guides/pagination.mdx:93]: I think we should include how to query for this data too. For instance, how they can get the `fetchMore` function like in the blueprint:

https://github.com/wpengine/atlas-blueprint-portfolio/blob/f6d4d5525711e307392b0f25501806b7486270d7/pages/posts/index.js#L22-L24

File: internal/faustjs.org/docs/guides/pagination.mdx (hunk 2 above)
Comment (Fran-A-Dev) [reply internal/faustjs.org/docs/guides/pagination.mdx:16]: Good call

File: internal/faustjs.org/docs/guides/pagination.mdx (hunk 6 above)
Comment (Fran-A-Dev) [reply internal/faustjs.org/docs/guides/pagination.mdx:93]: Good call out on this, would you want me to specifically use the Blue Print example or can I use my raw Faust.js example?

**Review by Fran-A-Dev — COMMENTED** [review]
Updated edits from Blake's feedback

**Review by blakewilson — APPROVED** [review]
LGTM 👍 Thanks Fran!

**Pull request merged**